x-pack/plugins/canvas/public/components/workpad_header/control_settings/custom_interval.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { EuiFlexGroup, EuiFlexItem, EuiFormRow, EuiButton, EuiFieldText } from '@elastic/eui';
+
+const getRefreshInterval = (val = '') => {
+  // if it's a number, just use it directly
+  if (!isNaN(Number(val))) {
+    return val;
+  }
+
+  // if it's a string, try to parse out the shorthand duration value
+  const match = String(val).match(/^([0-9]{1,})([hmsd])$/);
+
+  // TODO: do something better with improper input, like show an error...
+  if (!match) {
+    return;
+  }
+
+  switch (match[2]) {
+    case 's':
+      return match[1] * 1000;
+    case 'm':
+      return match[1] * 1000 * 60;
+    case 'h':
+      return match[1] * 1000 * 60 * 60;
+    case 'd':
+      return match[1] * 1000 * 60 * 60 * 24;
+  }
+};
+
+export const CustomInterval = ({ gutterSize, buttonSize, onSubmit, defaultValue }) => {
+  const [customInterval, setCustomInterval] = useState(defaultValue);
+
+  const handleChange = ev => setCustomInterval(ev.target.value);
+
+  return (
+    <form
+      onSubmit={ev => {
+        ev.preventDefault();
+        onSubmit(getRefreshInterval(customInterval));
+      }}
+    >
+      <EuiFlexGroup gutterSize={gutterSize}>
+        <EuiFlexItem>
+          <EuiFormRow
+            label="Set a custom interval"
+            helpText="Use shorthand notation, like 30s, 10m, or 1h"
+            compressed
+          >
+            <EuiFieldText value={customInterval} onChange={handleChange} />
```

Comment:
Consider showing this with `isInvalid={!match}`... that would require this `SFC` to be stateful, but that's ok.

Reply:
![May-20-2019 10-52-00](https://user-images.githubusercontent.com/404731/58041600-55fd1d00-7aed-11e9-9803-f95dddeac114.gif)

As of dd9d3b0abf, it now shows invalid input and disables the form.